operator/src/main/java/org/projectnessie/operator/resource/NessieImage.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+public class NessieImage {
+
+  public enum PullPolicy {
+    Always,
+    Never,
+    IfNotPresent
+  }
+
+  private PullPolicy pullPolicy = PullPolicy.IfNotPresent;
+  private String name = "projectnessie/nessie:latest";
```

Comment:
Is it some "standard" to use this notion including the tag, or should we maybe add another "version" property?

Reply:
It's up to debate: one single field is the easiest for users, and the most flexible approach. On the other end of the spectrum, some operators define complex structs like this:

https://github.com/k8ssandra/k8ssandra-operator/blob/main/pkg/images/images.go#L18-L49

It's more detailed, but requires from the user to "reason" about how the full image name is decomposed.

For now I think this is fine, we can "split" later if needed.